src/main/kotlin/org/typedb/typeql/plugin/jetbrains/psi/PsiTypeQLUtils.kt

```diff
@@ -0,0 +1,105 @@
+package org.typedb.typeql.plugin.jetbrains.psi
+
+import com.intellij.ide.scratch.ScratchUtil
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.search.FileTypeIndex
+import com.intellij.psi.search.GlobalSearchScope
+import com.intellij.psi.util.PsiTreeUtil
+import org.antlr.intellij.adaptor.lexer.RuleIElementType
+import org.typedb.typeql.plugin.jetbrains.TypeQLFileType
+import org.typedb.typeql.plugin.jetbrains.TypeQLLanguage
+import org.typedb.typeql.plugin.jetbrains.psi.constraint.PsiTypeQLRelatesType
+import org.typedb.typeql.plugin.jetbrains.usage.TypeQLDeclarationFinder
+import java.util.function.Consumer
+import java.util.stream.Collectors
+
+/**
+ * @author [Brandon Fergerson](mailto:bfergerson@apache.org)
+ */
+object PsiTypeQLUtils {
```

Comment:
Thanks, this works for now. I agree with 'Util's hate and normally points to a poor architecture. However, since this isn't critical infrastructure right now we can leave it as is